app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
```

Comment:
Prefer collecting all of the labels up front, then telling the GithubService to add all of the labels at once.

Also, previously, we had a method that would add only the labels that were a) new and b) existing.  This was because the GitHub API would create a label even if it didn't exist, which is not something we wanted it to do.  I'm not sure where that method went...I assume it was removed as part of the last refactoring?